buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -714,20 +714,95 @@ public static ByteBuf readBytes(ByteBufAllocator alloc, ByteBuf buffer, int leng
         }
     }
 
-    static int lastIndexOf(AbstractByteBuf buffer, int fromIndex, int toIndex, byte value) {
+    static int lastIndexOf(final AbstractByteBuf buffer, int fromIndex, final int toIndex, final byte value) {
         assert fromIndex > toIndex;
         final int capacity = buffer.capacity();
         fromIndex = Math.min(fromIndex, capacity);
-        if (fromIndex < 0 || capacity == 0) {
+        if (fromIndex <= 0) { // fromIndex is the exclusive upper bound.
             return -1;
         }
-        buffer.checkIndex(toIndex, fromIndex - toIndex);
+        final int length = fromIndex - toIndex;
+        buffer.checkIndex(toIndex, length);
+        if (!PlatformDependent.isUnaligned()) {
+            return linearLastIndexOf(buffer, fromIndex, toIndex, value);
+        }
+        final int longCount = length >>> 3;
+        int offset = fromIndex;
+        if (longCount > 0) {
+            final ByteOrder nativeOrder = ByteOrder.nativeOrder();
+            final boolean isNative = nativeOrder == buffer.order();
+            final boolean useLE = nativeOrder == ByteOrder.LITTLE_ENDIAN;
+            final long pattern = SWARUtil.compilePattern(value);
+            for (int i = 0; i < longCount; i++) {
+                // use the faster available getLong
+                final long word = useLE? buffer._getLongLE(offset - Long.BYTES)
```

Comment:
In the if-clause, you could do an initial `Long.BYTES` subtraction to avoid this computation in the loop.

Reply:
Moved it :D thanks!